services/blockchain-indexer/shared/indexer/genesisBlock.js

```diff
@@ -0,0 +1,112 @@
+/*
+ * LiskHQ/lisk-service
+ * Copyright © 2023 Lisk Foundation
+ *
+ * See the LICENSE file at the top-level directory of this distribution
+ * for licensing information.
+ *
+ * Unless otherwise agreed in a custom licensing agreement with the Lisk Foundation,
+ * no part of this software, including this file, may be copied, modified,
+ * propagated, or distributed except according to the terms contained in the
+ * LICENSE file.
+ *
+ * Removal or modification of this copyright notice is prohibited.
+ *
+ */
+const { MODULE } = require('../constants');
+const { updateTotalLockedAmounts } = require('../utils/blockchainIndex');
+const { requestConnector } = require('../utils/request');
+const requestAll = require('../utils/requestAll');
+
+const {
+	updateTotalStake,
+	updateTotalSelfStake,
+} = require('./transactionProcessor/pos/stake');
+
+const MODULE_SUB_STORE = Object.freeze({
+	TOKEN: {
+		USER: 'userSubstore',
+	},
+	POS: {
+		STAKERS: 'stakers',
+	},
+});
+
+const indexTokenModuleAssets = async (dbTrx) => {
+	const genesisBlockAssetsLength = await requestConnector(
+		'getGenesisAssetsLength',
+		{ module: MODULE.TOKEN, subStore: MODULE_SUB_STORE.TOKEN.USER },
+	);
+	const totalUsers = genesisBlockAssetsLength[MODULE.TOKEN][MODULE_SUB_STORE.TOKEN.USER];
+
+	const tokenModuleData = await requestAll(
+		requestConnector,
+		'getGenesisAssetByModule',
+		{ module: MODULE.TOKEN, subStore: MODULE_SUB_STORE.TOKEN.USER },
+		totalUsers,
+	);
+	const userSubStoreInfos = tokenModuleData[MODULE_SUB_STORE.TOKEN.USER];
```

Comment:
Just an suggestion, this can be updated to:
```suggestion
	const { userSubstore: userSubStoreInfos } = await requestAll(
		requestConnector,
		'getGenesisAssetByModule',
		{ module: MODULE.TOKEN, subStore: MODULE_SUB_STORE.TOKEN.USER },
		totalUsers,
	);
```

Reply:
Thanks for the suggestion. I would prefer keeping keeping it as we are using the constant to access the property instead of hardcoding.